include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_kernels.h

```diff
@@ -0,0 +1,687 @@
+        _ONEDPL_PRAGMA_UNROLL
+        for (::std::uint32_t __grf_offset = 0; __grf_offset < __bin_count * __stages_per_block; __grf_offset += __data_per_step)
+        {
+            ::std::uint32_t slm_offset = __stage_block_start * __bin_count + __grf_offset;
+            __dpl_esimd_ns::simd<::std::uint32_t, __data_per_step> __slm_byte_offsets(slm_offset * sizeof(_GlobalHistT), sizeof(_GlobalHistT));
+            __dpl_esimd_ens::lsc_slm_atomic_update<__dpl_esimd_ns::atomic_op::add, _GlobalHistT, __data_per_step>(
+                __slm_byte_offsets, __state_hist_grf.template select<__data_per_step, 1>(__grf_offset), 1);
+        }
+        __dpl_esimd_ns::barrier();
+    }
+
+    // 4. Reduce group-local historgrams from SLM into global histograms in global memory
+    __dpl_esimd_ns::simd<_GlobalHistT, __group_hist_size> __group_hist =
+        __utils::__block_load_slm<_GlobalHistT, __group_hist_size>(__local_id * __group_hist_size *
+                                                                    sizeof(_GlobalHistT));
+    __dpl_esimd_ns::simd<::std::uint32_t, __group_hist_size> __byte_offsets(0, sizeof(_GlobalHistT));
+    __dpl_esimd_ens::lsc_atomic_update<__dpl_esimd_ns::atomic_op::add>(__p_global_offset + __local_id * __group_hist_size,
+                                                      __byte_offsets, __group_hist,
+                                                      __dpl_esimd_ns::simd_mask<__group_hist_size>(1));
+}
+
+template <bool __is_ascending, ::std::uint8_t __radix_bits,
+          ::std::uint16_t __data_per_work_item, ::std::uint16_t __work_group_size,
+          typename _InRngPack, typename _OutRngPack>
+struct __radix_sort_onesweep_kernel
+{
+    using _LocOffsetT = ::std::uint16_t;
+    using _GlobOffsetT = ::std::uint32_t;
+
+    using _KeyT = typename _InRngPack::_KeyT;
+    using _ValT = typename _InRngPack::_ValT;
+    static constexpr bool __has_values = !::std::is_void_v<_ValT>;
+
+    static constexpr ::std::uint32_t __bin_count = 1 << __radix_bits;
+
+    template<typename _T, ::std::uint16_t _N>
+    using _SimdT =  __dpl_esimd_ns::simd<_T, _N>;
+
+    using _LocOffsetSimdT = _SimdT<_LocOffsetT, __data_per_work_item>;
+    using _GlobOffsetSimdT = _SimdT<_GlobOffsetT, __data_per_work_item>;
+    using _LocHistT = _SimdT<_LocOffsetT, __bin_count>;
+    using _GlobHistT = _SimdT<_GlobOffsetT, __bin_count>;
+
+    static constexpr ::std::uint32_t __bit_count = sizeof(_KeyT) * 8;
+    static constexpr _LocOffsetT __mask = __bin_count - 1;
+    static constexpr ::std::uint32_t __hist_stride = __bin_count * sizeof(_LocOffsetT);
+    static constexpr ::std::uint32_t __work_item_all_hists_size = __work_group_size * __hist_stride;
+    static constexpr ::std::uint32_t __group_hist_size = __hist_stride;
+    static constexpr ::std::uint32_t __global_hist_size = __bin_count * sizeof(_GlobOffsetT);
+
+    static constexpr ::std::uint32_t __calc_reorder_slm_size()
```

Comment:
Probably, it makes sense to define that contexpr value as a static constexpr member?
`static constexpr ::std::uint32_t  __reorder_slm_size = __work_group_size * __data_per_work_item * (sizeof(_KeyT) + (__has_values ? sizeof(_ValT) : 0));`

Reply:
`_ValT` becomes `void` if this is a key-only sort. `sizeof(void)` will produce a compilation error in this expression.